core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -542,8 +546,7 @@ trait DerivationSchema[R] {
   private def getDescription[Typeclass[_], Type](ctx: ReadOnlyParam[Typeclass, Type]): Option[String] =
     getDescription(ctx.annotations)
 
-  implicit def gen[T]: Typeclass[T] = macro Magnolia.gen[T]
-
+  implicit def gen[T]: Exported[Typeclass[T]] = macro ExportedMagnolia.exportedMagnolia[Typeclass, T]
```

Comment:
How about calling this differently and make `gen` return the same thing as before? So that we don't need to change every existing call.

Reply:
Have some experiments with this.

1. Old method
```scala
implicit def gen[T]: Typeclass[T] = Magnolia.gen[T]
```
Will pollute implicit scope when imported.

2. Without implicit
```scala
def gen[T]: Typeclass[T] = Magnolia.gen[T]
```
Different semantic: magnolia disables recursive generation without `implicit` keyword
https://github.com/propensive/magnolia/blob/master/core/shared/src/main/scala/magnolia.scala#L249-L251

3. Using implicit
```scala
def gen[T](implicit e: Exported[Typeclass[T]]) = e.instance
```
Works with default schema (`Schema.gen[MyClass]`) but not work with custom schemas like this:
```scala
object mySchema extends GenericSchema[Blocking]
mySchema.gen[MyClass] // this will fail 
```

I will experiment with this a bit more on weekend.

Reply:
What if you renamed the implicit `gen` into e.g. `autogen` and added a new `def gen` that is not implicit and just a shortcut to avoid calling `.instance`?

Reply:
I think the 3rd one is acceptable, actually the caliban docs say you should `import mySchema._` when you use this pattern, which makes the implicit in scope. I opened https://github.com/ghostdogpr/caliban/pull/658 to get this in.